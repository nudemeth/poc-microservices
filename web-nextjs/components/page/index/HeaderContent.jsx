import React from "react"
import PropTypes from "prop-types"
import Input from "@material-ui/core/Input"
import InputAdornment from "@material-ui/core/InputAdornment"
import Icon from "@material-ui/core/Icon"
import { withStyles } from "@material-ui/core/styles"

const styles = () => ({
    wrapper: {
        display: "flex",
        justifyContent: "flex-end"
    },
    textField: {
        color: "#fff",
        backgroundColor: "rgba(255,255,255,0.2)"
    },
    underline: {
        "&:before": {
            height: "100%",
            backgroundColor: "rgba(255,255,255,0.42)"
        },
        "&:hover": {
            "&:not(.MuiInput-disabled)": {
                "&:before": {
                    backgroundColor: "rgba(255,255,255,0.6)"
                }
            }
        },
        "&:after": {
            backgroundColor: "rgba(255,255,255,1)"
        }
    },
    adorment: {
        marginLeft: 8
    }
})

class HeaderContent extends React.Component {
    constructor(props) {
        super(props)
    }

    static propTypes = {
        classes: PropTypes.object.isRequired,
        theme: PropTypes.object.isRequired
    }

    render() {
        return (
            <div className={this.props.classes.wrapper}>
                <Input
                    id="search"
                    type="search"
                    classes={{
                        root: this.props.classes.textField,
                        underline: this.props.classes.underline
                    }}
                    startAdornment={
                        <InputAdornment position="start" className={this.props.classes.adorment}>
                            <Icon>search</Icon>
                        </InputAdornment>
                    }
                />
            </div>
        )
    }
}

export default withStyles(styles, { withTheme: true })(HeaderContent)
export { HeaderContent }