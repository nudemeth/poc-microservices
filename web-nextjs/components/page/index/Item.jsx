import React from 'react'
import PropTypes from 'prop-types'
import GridListTile from '@material-ui/core/GridListTile'
import GridListTileBar from '@material-ui/core/GridListTileBar'
import Icon from '@material-ui/core/Icon'
import IconButton from '@material-ui/core/IconButton'
import { withStyles } from '@material-ui/core/styles'
import config from '../../../config'

const styles = theme => ({
    itemTile: {
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: '#fff'
    },
    itemRoot: {
        [theme.breakpoints.up('lg')]: {
            width: '33.3333%',
            padding: 12
        },
        [theme.breakpoints.down('md')]: {
            width: '50%',
            padding: 8
        },
        [theme.breakpoints.down('xs')]: {
            width: '100%',
            padding: 2
        }
    },
    button: {
        color: '#fff'
    }
})

class Item extends React.Component {
    constructor(props) {
        super(props)
    }

    static propTypes = {
        classes: PropTypes.object.isRequired,
        theme: PropTypes.object.isRequired,
        item: PropTypes.object.isRequired
    }

    render() {
        const { classes, item } = this.props
        const imageUrl = `${config.api.catalog.uri}items/${item.id}/img`
        return (
            <GridListTile classes={{root: classes.itemRoot, tile: classes.itemTile}}>
                <img src={imageUrl} alt={item.name} />
                <GridListTileBar
                    title={item.name}
                    actionIcon={
                        <IconButton className={classes.button} aria-label='Add to shopping cart'>
                            <Icon>add_shopping_cart</Icon>
                        </IconButton>
                    }
                />
            </GridListTile>
        )
    }
}

export default withStyles(styles, { withTheme: true })(Item)
export { Item }