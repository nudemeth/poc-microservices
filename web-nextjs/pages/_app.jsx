import React from 'react'
import App, { Container } from 'next/app'
import Router from 'next/router'
import { MuiThemeProvider } from '@material-ui/core/styles'
import CssBaseline from '@material-ui/core/CssBaseline'
import JssProvider from 'react-jss/lib/JssProvider'
import { Provider } from 'react-redux'
import withRedux from 'next-redux-wrapper'
import withReduxSaga from 'next-redux-saga'
import getPageContext from '../components/common/getPageContext'
import configureStore from '../store/store'
import { storeAccessToken, storeAuthSites } from '../actions/identity.actions'
import config from '../config'

class MyApp extends App {
    constructor(props) {
        super(props)
        this.pageContext = getPageContext()
    }

    static async getInitialProps ({ Component, ctx }) {
        let pageProps = {}

        if (ctx.isServer) {
            ctx.store.dispatch(storeAccessToken(ctx.query.accessToken))
            ctx.store.dispatch(storeAuthSites(ctx.query.sites))
        }

        if (Component.getInitialProps) {
            pageProps = await Component.getInitialProps({ ctx })
        }

        if (!ctx.isServer) {
            return { pageProps }
        }

        // Wait Saga task to be done on server side
        await ctx.store.waitSagaTaskDone(ctx.isServer)

        const accessToken = ctx.store.getState().identityReducer.accessToken

        if (this.isRedirect(accessToken, ctx.pathname)) {
            this.redirect(ctx.res)
        }

        return { pageProps }
    }

    static redirect(res) {
        if (!res) {
            Router.push('/login')
            return
        }

        res.writeHead(302, {
            Location: '/login'
        })
        res.end()
    }

    static isRedirect(accessToken, pathname) {
        return !accessToken && !config.noAuthPaths.includes(pathname)
    }

    componentDidMount() {
        // Remove the server-side injected CSS.
        const jssStyles = document.querySelector('#jss-server-side')
        if (jssStyles && jssStyles.parentNode) {
            jssStyles.parentNode.removeChild(jssStyles)
        }
    }

    componentDidUpdate() {
        const accessToken = this.props.store.getState().identityReducer.accessToken
        const pathname = this.props.router.pathname
        if (MyApp.isRedirect(accessToken, pathname)) {
            this.props.router.push('/login')
        }
    }

    render() {
        const { Component, pageProps, store } = this.props
        return (
            <Container>
                <Provider store={store}>
                    {/* Wrap every page in Jss and Theme providers */}
                    <JssProvider
                        registry={this.pageContext.sheetsRegistry}
                        generateClassName={this.pageContext.generateClassName}
                    >
                        {/* MuiThemeProvider makes the theme available down the React
                        tree thanks to React context. */}
                        <MuiThemeProvider
                            theme={this.pageContext.theme}
                            sheetsManager={this.pageContext.sheetsManager}
                        >
                            {/* CssBaseline kickstart an elegant, consistent, and simple baseline to build upon. */}
                            <CssBaseline />
                            {/* Pass pageContext to the _document though the renderPage enhancer
                                to render collected styles on server side. */}
                            <Component pageContext={this.pageContext} {...pageProps} />
                        </MuiThemeProvider>
                    </JssProvider>
                </Provider>
            </Container>
        )
    }
}

export default withRedux(configureStore)(withReduxSaga(MyApp))