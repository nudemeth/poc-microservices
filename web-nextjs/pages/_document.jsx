import React from 'react'
import Document, { Head, Main, NextScript } from 'next/document'
import { JssProvider }  from 'react-jss'
import getCssContext from '../components/common/getCssContext'

class DefaultDocument extends Document {
    static getInitialProps({ renderPage }) {
        // Resolution order
        //
        // On the server:
        // 1. page.getInitialProps
        // 2. document.getInitialProps
        // 3. page.render
        // 4. document.render
        //
        // On the server with error:
        // 2. document.getInitialProps
        // 3. page.render
        // 4. document.render
        //
        // On the client
        // 1. page.getInitialProps
        // 3. page.render

        // Get the context of the page to collected side effects.
        const cssContext = getCssContext()
        const page = (Page) => (props) => {
            const jss = <JssProvider
                registry={cssContext.sheetsRegistry}
                generateClassName={cssContext.generateClassName}
            >
                <Page cssContext={cssContext} {...props} />
            </JssProvider>
            return jss
        }
        const renderedPage = renderPage(page)

        return {
            ...renderedPage,
            cssContext,
            styles: (
                <style id='jss-server-side' dangerouslySetInnerHTML={{ __html: cssContext.sheetsRegistry.toString() }} />
            )
        }
    }

    render() {
        return (
            <html>
                <Head>
                    <meta charSet='utf-8' />
                    <meta name='author' content='nudemeth' />
                    {/* Use minimum-scale=1 to enable GPU rasterization */}
                    <meta
                        name='viewport'
                        content={
                            'user-scalable=0, initial-scale=1, ' +
                            'minimum-scale=1, width=device-width, height=device-height'
                        }
                    />
                    {/* PWA primary color */}
                    <meta name='theme-color' content={this.props.cssContext.theme.palette.primary[500]} />
                    <link href='https://fonts.googleapis.com/css?family=Roboto:300,400,500' rel='stylesheet' type='text/css' />
                    <link href='https://fonts.googleapis.com/icon?family=Material+Icons' rel='stylesheet' type='text/css' />
                    <link href='./static/css/style.css' rel='stylesheet' type='text/css' />
                </Head>
                <body>
                    <Main />
                    <NextScript />
                </body>
            </html>
        )
    }
}

export default DefaultDocument