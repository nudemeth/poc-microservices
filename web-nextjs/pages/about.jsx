import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import Container from '../components/layout/Container'
import { initGreeting, updateGreeting } from '../actions/about.actions'

class About extends React.Component {
    constructor(props) {
        super(props)
    }

    static async getInitialProps({ ctx: { store } }) {
        store.dispatch(initGreeting('This is from server'))
        return {}
    }

    static propTypes = {
        dispatch: PropTypes.func.isRequired,
        greeting: PropTypes.string.isRequired
    }

    componentDidMount() {
        this.props.dispatch(updateGreeting())
    }

    render() {
        return (
            <Container title='About' header={<div/>}>
                <h1>This is About Page: {this.props.greeting}</h1>
            </Container>
        )
    }
}

const mapStateToProps = ({ aboutReducer: { greeting }}) => ({ greeting })

export default connect(mapStateToProps)(About)
export { About }