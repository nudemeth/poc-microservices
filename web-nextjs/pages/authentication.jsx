import React from 'react'

class Authentication extends React.Component {
    componentDidMount() {
        if (typeof window != 'undefined') {
            window.opener.location.reload()
            window.close()
        }
    }

    render() {
        return null
    }
}

export default Authentication
export { Authentication }