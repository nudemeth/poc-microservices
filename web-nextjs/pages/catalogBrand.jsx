import React from "react"
import PropTypes from "prop-types"
import Container from "../components/layout/Container"
import { loadCatalogBrands } from "../actions/catalogBrand.actions"
import { withReduxSaga } from "../store/store"
import CatalogBrandList from "../components/page/catalogBrand/CatalogBrandList"

class CatalogBrand extends React.Component {
    constructor(props) {
        super(props)
    }

    static async getInitialProps({ store }) {
        store.dispatch(loadCatalogBrands())
    }

    static propTypes = {
        dispatch: PropTypes.func.isRequired
    }

    componentDidMount() {
        this.props.dispatch(loadCatalogBrands())
    }

    render() {
        return (
            <Container title="Catalog Brand">
                <CatalogBrandList/>
            </Container>
        )
    }
}

export default withReduxSaga()(CatalogBrand)
export { CatalogBrand }