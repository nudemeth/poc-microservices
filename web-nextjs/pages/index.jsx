import React from 'react'
import PropTypes from 'prop-types'
import Router from 'next/router'
import { connect } from 'react-redux'
import Container from '../components/layout/Container'
import Catalog from '../components/page/index/Catalog'
import HeaderContent from '../components/page/index/HeaderContent'
import { loadItems } from '../actions/catalog.actions'

class Index extends React.Component {
    constructor(props) {
        super(props)
    }

    static async getInitialProps({ ctx: { store, res, query } }) {
        if (!this.isRedirect(query)) {
            store.dispatch(loadItems())
            return
        }

        this.redirect(res)
    }

    static propTypes = {
        dispatch: PropTypes.func.isRequired
    }

    static redirect(res) {
        if (res) {
            res.writeHead(302, {
                Location: '/login'
            })
            res.end()
        } else {
            Router.push('/login')
        }
    }

    static isRedirect(query) {
        return !query.accessToken
    }

    componentDidMount() {
        this.props.dispatch(loadItems())
    }

    render() {
        return (
            <Container title='Index' header={<HeaderContent/>}>
                <Catalog/>
            </Container>
        )
    }
}

export default connect()(Index)
export { Index }