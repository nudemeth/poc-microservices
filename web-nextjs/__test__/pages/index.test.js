import React from 'react'
import Router from 'next/router'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import sinon from 'sinon'
import { Index } from '../../pages/index'

const mockedRouter = { prefetch: () => {} }
Router.router = mockedRouter

describe('Index page', () => {
    it('Should show "Index Page: Prodcuts Page" in Index page', () => {
        const spyDispatch = sinon.spy()
        shallow(<Index dispatch={spyDispatch} title='Home' />)
        sinon.assert.calledWith(spyDispatch, { type: 'LOAD_ITEMS' })
    })
})

jest.mock('../../components/page/index/Catalog', () => {
    return () => '<div>Mocked Catalog</div>'
})

describe('Index page with Snapshot Testing', () => {
    it('Should show "Index Page: Prodcuts Page" in Index page', () => {
        const spyDispatch = sinon.spy()
        const wrapper = shallow(<Index dispatch={spyDispatch} title='Home' />)
        const tree = toJSON(wrapper)
        expect(tree).toMatchSnapshot()
    })
})