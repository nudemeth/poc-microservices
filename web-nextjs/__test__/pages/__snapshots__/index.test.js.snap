// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index page with Snapshot Testing Should show "Index Page: Prodcuts Page" in Index page 1`] = `
<WithStyles(Container)
  header={<WithStyles(HeaderContent) />}
  title="Index"
>
  <Component />
</WithStyles(Container)>
`;
